https://colab.research.google.com/drive/1P9iQHtUH5KUbTVtp8B4-AopZzEEPE0lw?usp=sharing#scrollTo=4YS7XI2bZTyz

https://github.com/aub-mind/arabert

In [ ]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [ ]:
!pip -q install transformers==4.2.1

     |████████████████████████████████| 1.8 MB 8.0 MB/s 
     |████████████████████████████████| 895 kB 52.2 MB/s 
     |████████████████████████████████| 2.9 MB 42.6 MB/s 


In [ ]:
#Import pipeline
from transformers import pipeline
from transformers import AutoTokenizer,AutoModelForSequenceClassification
from sklearn.metrics import classification_report, accuracy_score, f1_score, confusion_matrix, precision_score , recall_score

import torch
import pandas as pd

In [ ]:
#Import pipeline
from transformers import pipeline
Model="/content/drive/MyDrive/Arabic_FakeNews/Model/Balance/SavedModels/QaRiB"

tokenizer = AutoTokenizer.from_pretrained(Model)
model = AutoModelForSequenceClassification.from_pretrained(Model)

In [ ]:
ANS_df=pd.read_csv('/content/drive/MyDrive/Arabic_FakeNews/Model/ANS/Data/PPV2dev_news.csv')

In [ ]:
y_pred_news = []
for news in ANS_df.text:
  paraphrase = tokenizer(news, return_tensors="pt")
  paraphrase_classification_logits=model(**paraphrase).logits
  paraphrase_results = torch.softmax(paraphrase_classification_logits, dim=1).tolist()[0]

  if(paraphrase_results[0]> paraphrase_results[1]):
    y_pred_news.append(0)
  else:
    y_pred_news.append(1)


In [ ]:
y_test=ANS_df.label
print("Accuracy", accuracy_score(y_test, y_pred_news))
print("\nConfusion matrix\n",confusion_matrix(y_test, y_pred_news))
print("\nClassification report\n",classification_report(y_test, y_pred_news))
print("\n f1_score report ",f1_score(y_test, y_pred_news,average='macro'))
print("\n macro_precision  ",precision_score(y_test, y_pred_news,average='macro'))
print("\n macro_recall  ",recall_score(y_test, y_pred_news,average='macro'))

Accuracy 0.4955849889624724

Confusion matrix
 [[365 251]
 [206  84]]

Classification report
               precision    recall  f1-score   support

           0       0.64      0.59      0.61       616
           1       0.25      0.29      0.27       290

    accuracy                           0.50       906
   macro avg       0.44      0.44      0.44       906
weighted avg       0.51      0.50      0.50       906


 f1_score report  0.4418978938500421

 macro_precision   0.44498784536163316

 macro_recall   0.44109381997313035


Accuracy 0.5452538631346578

Confusion matrix
 [[437 179]
 [233  57]]

Classification report
               precision    recall  f1-score   support

           0       0.65      0.71      0.68       616
           1       0.24      0.20      0.22       290

    accuracy                           0.55       906
   macro avg       0.45      0.45      0.45       906
weighted avg       0.52      0.55      0.53       906


 f1_score report
 0.44817839381700564

<!-- Accuracy 0.5452538631346578

Confusion matrix
 [[437 179]
 [233  57]]

Classification report
               precision    recall  f1-score   support

           0       0.65      0.71      0.68       616
           1       0.24      0.20      0.22       290

    accuracy                           0.55       906
   macro avg       0.45      0.45      0.45       906
weighted avg       0.52      0.55      0.53       906


 f1_score report
 0.44817839381700564 -->